src/anyio/pytest_plugin.py

```diff
@@ -82,15 +116,23 @@ def wrapper(*args, anyio_backend, **kwargs):  # type: ignore[no-untyped-def]
             else:
                 yield runner.run_fixture(func, kwargs)
 
-    # Only apply this to coroutine functions and async generator functions in requests
-    # that involve the anyio_backend fixture
-    func = fixturedef.func
-    if isasyncgenfunction(func) or iscoroutinefunction(func):
-        if "anyio_backend" in request.fixturenames:
-            has_backend_arg = "anyio_backend" in fixturedef.argnames
-            fixturedef.func = wrapper
-            if not has_backend_arg:
-                fixturedef.argnames += ("anyio_backend",)
+    with ExitStack() as stack:
+        # Only apply this to coroutine functions and async generator functions in requests
+        # that involve the anyio_backend fixture
+        func = _resolve_fixture_function(fixturedef, request)
+        if isasyncgenfunction(func) or iscoroutinefunction(func):
+            if "anyio_backend" in request.fixturenames:
+                has_backend_arg = "anyio_backend" in fixturedef.argnames
+                original_func = fixturedef.func
+                fixturedef.func = wrapper
+                stack.callback(setattr, fixturedef, "func", original_func)
```

Comment:
this should fix a case where anyio adopts a fixture and then subsequent tests use it